### ISSUE TITLE
Allow specify relative path in `transform.conf.json`

### ISSUE THREAD

**Issue by yaquawa** [aws-amplify/amplify-cli#7484]
### Is this related to a new or existing Amplify category?

api

### Describe the feature you'd like to request

currently if I want to use a custom transformer, I have to create a npm package or specify a absolute path from my local machine:

```json
{
    "transformers": [
        "npm-package-name",
        "file:///Users/XX/Documents/YY/src/infrastructures/amplify/backend/api/PP/transformers/myTransformer.js"
    ]
}
```

I don't want to publish a custom npm package for my private project, but if I use a absolute path, other team member won't be able to run the `amplify push` because the absolute path is bound to my local machine.

So the third option could be using "workspace" in package.json, but I don't want to do that for some reasons.

### Describe the solution you'd like

use `../` to represent the parent directory or use `<root>` to repuresent the project root.

```json
{
    "transformers": [
        "file://../transformers/myTransformer.js"
        "<root>/transformers/myTransformer.js"
    ]
}
```

### Describe alternatives you've considered

xx

### Additional context

Related code:

https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-provider-awscloudformation/src/transform-graphql-schema.ts#L102

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by attilah** [comment]
@yaquawa in this [re:Invent 2019 workshop](https://github.com/amazon-archives/aws-reinvent-2019-mobile-workshops/tree/master/MOB405) there is a sample on how to build and reference a custom transformer with a relative path, check if that works for you and get back to us.

**Comment by yaquawa** [comment]
Thanks @attilah ! Could you point out where is the sample located at?

**Comment by attilah** [comment]
At the link above ^

**Comment by yaquawa** [comment]
@attilah Do you mean this? https://github.com/amazon-archives/aws-reinvent-2019-mobile-workshops/blob/master/MOB405/01%20-%20Finished/validatortest/amplify/backend/api/validatortest/transform.conf.json

but this is not a relative path relative to the file right?

**Comment by attilah** [comment]
No, it is this line: https://github.com/amazon-archives/aws-reinvent-2019-mobile-workshops/blob/master/MOB405/01%20-%20Finished/validatortest/package.json#L13 that adds the custom transformer - unpublished - package to the project's dependency and since this way it gets installed to the amplify project's `node_modules`, the relative path `./<package-name>` will resolve to that when transformer is looking for the package.

This is the transformer loading code: https://github.com/aws-amplify/amplify-cli/blob/5645c711844b5f9a9af248a31e5e264e0ba1bcdf/packages/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts#L67-L102 if you want to see how it is resolving the `transform.conf.json` referenced custom transformers.

**Comment by yaquawa** [comment]
Thanks @attilah , but I have already read the source code, and found that it could not locate a transformer js file relative to the file `transform.conf.json`, that why I created this issue(feature request).

**Comment by yaquawa** [comment]
What I want to do is keep the custom transformers in the same directory of `transform.conf.json`, I don't want to create a lots of directories under my project root just for such a reason.

**Comment by attilah** [comment]
I leave it as an enhancement then, that is not a use case that we support Today.

**Comment by yaquawa** [comment]
Got it, thanks @attilah
